src/wirecraft/client/ui/cable.py

```diff
@@ -27,26 +35,52 @@ def update_position(self, camera: Camera, resolution: Resolution) -> None:
-        if self.start_world is None:
-            # self.start_world = camera.screen_to_world(self.start, screen_size)  # TODO: use int
-            pass
-        if self.ended and self.end_world is None:
-            # self.end_world = camera.screen_to_world(self.end, screen_size)  # TODO: use int
-            pass
 
-        # Update start point screen position
-        # self.start = camera.world_to_screen(self.start_world, screen_size)  # TODO: use int
+    # this function looks unnecessary as it works fine without it but to be confirmed
+    # screen_size = resolution.size  # Get the tuple from Resolution object
+    # screen_size = resolution.size  # Get the tuple from Resolution object
 
-        # Update end point screen position only if cable is complete
-        if self.ended and self.end_world is not None:
-            # self.end = camera.world_to_screen(self.end_world, screen_size)  # TODO: use int
-            pass
+    #
+    ## First time update: calculate and store world coordinates
+    # if self.start_world is None:
+    #    self.start_world = camera.screen_to_world(self.start, screen_size)
+    # if self.ended and self.end_world is None:
+    #    self.end_world = camera.screen_to_world(self.end, screen_size)
+    #
+    ## Update start point screen position
+    # self.start = camera.world_to_screen(self.start_world, screen_size)
+    #
+    ## Update end point screen position only if cable is complete
+    # if self.ended and self.end_world is not None:
+    #    self.end = camera.world_to_screen(self.end_world, screen_size)
 
     def update_zoom(self, camera: Camera) -> None:
         pass
 
     def draw(self, surface: pygame.Surface, camera: Camera, resolution: Resolution) -> None:
-        pygame.draw.line(surface, RED, self.start, self.end, 5)
+        # TODO fix position of the cable
+        pygame.draw.line(
+            surface,
+            RED,
+            camera.world_to_screen(
+                (
+                    server_var.get().get_device_pos(self.id_device1)[0]
+                    + server_var.get().get_port_pos(self.port_device1, self.id_device1)[0],
+                    server_var.get().get_device_pos(self.id_device1)[1]
+                    + server_var.get().get_port_pos(self.port_device1, self.id_device1)[1],
+                ),
+                resolution.size,
+            ),
+            camera.world_to_screen(
+                (
+                    server_var.get().get_device_pos(self.id_device2)[0]
+                    + server_var.get().get_port_pos(self.port_device2, self.id_device2)[0],
+                    server_var.get().get_device_pos(self.id_device2)[1]
+                    + server_var.get().get_port_pos(self.port_device2, self.id_device2)[1],
+                ),
+                resolution.size,
+            )
+            if self.id_device2 > 0
+            else pygame.mouse.get_pos(),
+            width=5,
+        )
```

Comment:
I'm terrified of this code.
btw HERE you should not use server_var.get()

The interface should be used. server_var should not be used except inside the ServerInterface and the exit function of the client.

Reply:
Boo :ghost:, honestly I don't really know how to refractor it (will look into it)
Good catch on server_var.get forgot to change it here

Reply:
I guess simply create named variables instead of set everything in function call would be a go improvement to starts with.